src/rpc/client_channel.rs

```diff
@@ -236,24 +232,18 @@ impl WebRTCClientChannel {
             loop {
                 let split_at = MAX_REQUEST_MESSAGE_PACKET_DATA_SIZE
                     .min(data.len())
-                    .min(usize::try_from(next_message_length).unwrap());
+                    .min(next_message_length);
                 let (to_send, remaining) = data.split_at(split_at);
-                next_message_length -= u32::try_from(split_at).unwrap();
+                next_message_length -= split_at;
                 let stream = stream.clone();
+                let eos = remaining.is_empty();
```

Comment:
We got bit here a bit by trying to be too close to `Go`. The Go logic doesn't seem to have a way of reliably saying that the stream is done, hence taking `eos` as an argument. We initially copied that in Rust, but it turns out we _do_ have a good way to determine if the stream is done or not, since we receive a complete vec of all the calls. Thus we can just check for that `vec` to be empty, removing the need for an `eos` value to ever be passed and reducing the risk of it being passed incorrectly.

Worth noting, the case where there is meaningful latency delay between messages being sent on the client side in a bidi streaming context remains a little uncertain. We seem to consistently receive all messages in a single vec, and thus can reliably infer when `eos` occurs. I'm not certain what will happen in cases where there's a heavy delay (is there blocking somewhere in rust to collect all the stream messages? will we end up making multiple `call` calls under the hood?), in part because (to my knowledge) we don't have any APIs that work that way yet. But if and when we do, some revisiting may be necessary. There's a comment to this effect in the file already.